common/djangoapps/student/management/commands/populate_is_marketable_user_attribute.py

```diff
@@ -0,0 +1,103 @@
+""" Management command to back-populate marketing emails opt-in for the user accounts. """
+
+import time
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+
+from common.djangoapps.student.models import UserAttribute
+from common.djangoapps.util.query import use_read_replica_if_available
+
+OLD_USER_ATTRIBUTE_NAME = 'marketing_emails_opt_in'
+NEW_USER_ATTRIBUTE_NAME = 'is_marketable'
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py lms populate_is_marketable_user_attribute
+    """
+    help = """
+        Creates a row in the UserAttribute table for all users in the platform.
+        This command back-populates the 'is_marketable' attribute in the
+        UserAttribute table for the user accounts.
+        """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--batch-delay',
+            type=float,
+            dest='batch_delay',
+            default=0.5,
+            help='Time delay in each iteration'
+        )
+        parser.add_argument(
+            '--batch-size',
+            type=int,
+            dest='batch_size',
+            default=5000,
+            help='Batch size'
+        )
+
+    def _get_user_attribute_queryset(self, user_attribute_id, batch_size):
+        """ Fetches the user attributes in batches """
+        self.stdout.write(
+            f'Fetching user attributes in batch starting from ID {user_attribute_id} with batch size {batch_size}'
+        )
+        query_set = UserAttribute.objects.filter(
+            id__gt=user_attribute_id,
+            name=OLD_USER_ATTRIBUTE_NAME
+        ).order_by('id')[:batch_size]
```

Comment:
Can we simplify it and use where query? 
```suggestion
        query_set = UserAttribute.objects.filter(
            id__gte=user_attribute_id,
            id_lt=user_attribute_id + batch_size,
            name=OLD_USER_ATTRIBUTE_NAME
        ).order_by('id')
```

Reply:
Also because of the use of `gt` instead of `gte` we will miss the 0th index user

Reply:
- Using `LIMIT` has no performance degradation, only `OFFSET` has on large tables
- autoincrement's default value is 1 (start index), so `gt` will work here